wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/repository/WalletRepository.kt

```diff
@@ -221,6 +232,58 @@ class WalletRepository(
         }
     }
 
+    private var isChainRefreshed = false
+    private suspend fun refreshChainDataWithTokens() {
+        try {
+            val chains = if (isChainRefreshed) {
```

Comment:
Can we make this as stateless as possible? :thinking:

Reply:
updated